### DISCUSSION TITLE
Before starting a remote session show legal terms which have to be checked (1, 2 or more terms)

### DISCUSSION THREAD

**Discussion by n1smithy** [rustdesk/rustdesk#5069] [Feature Request]
Hello folks,

I'm Smithy from Germany, this is my very first posting at github. I use Rustdesk as a remote supporting tool for my clients. I'm just running a very small one person business, so there's not much budget for "professional" remote support software.

However, it would be a very nice feature – exspecially for German/European Union users – to add custom terms of use for private customers (in German one very special and important legal term is called "Widerrufsbelehrung" and it's for "Fernabsatzverträge"). Those should be shown before starting a remote session (with no password access to the remote client).

One should be able to create custom texts with checkboxes (as many as one need), which must be checked after the supporter tries to get access to the client's computer but also before the remote session will be initiated. It would be great, if one can choose 2 ways. For example: First option is only one checkbox for remote support for b2b ("By checking the left box you assure that this is a business to business remote support session and that you do not act as a private customer. By checking the next checkbox you accept our legal terms for b2b support."). Second option would be 3 checkboxes for "Instructions on withdrawal", "Accepting the right of withdrawal expiring prematurely for this contract" and of course the legal conditions of the supporter. By writing one's full name and maybe one's e-mail address the client can accept the terms.

It would also be great, if the supporter could "build" one's own ad hoc remote support module and safe it in the executable file.

I know this is a complex feature, but it would be a damn good feature!

**Comment by dinger1986** [comment]
Could you not implement that with a web page and depending on their answer it doesnt connect?

**Reply by n1smithy** [reply]
Hi, dinger!

Well, I thought about that, and I think, that's for the first time the customer has to download rustdesk it would be okay. But once the software is on the computer – must commonly I put it on the desktop for easier access – there would be no questioning anymore. But as the EU wishes the terms must be accepted each and every time a contract is closed. At least it's in Germany that way.

**Reply by it-boerner** [reply]
But if they download any other software, then it is up to the person who downloaded the software from the Internet.

There is the provision regarding data protection, but he actively agrees that the connection must be established. The connection will not be established without consent.

Think of TV or other products, there is no reference to the GDPR either.

**Reply by n1smithy** [reply]
Hi, IT Börner,

it seem you are from Germany, too. After my English reply I will answer in German, too. It's easier for me to tell you about the law facts in German than in English.

However, it's not like buying a ware – even if it can be that someone buys hard- or software from me. In fact the European consumer protection directive forces a businessman to inform the customer in written form about his rights and the possibility to give them up to receive the service immediately always (!) before the contract is closed. For example: the consumer can cancel the contract 14 days after fullfilling the services even if everything was completely done.

The whole thing isn't about the software one use. It's about the service. And if you do not inform the customer about his rights before the contract is closed, he can legally deny paying for the service if he use his right for contradiction.

But there's a much bigger problem than this, because most customers pay their bills. You can be legally warned for unfair competition. This can be expensive!

Greetings!

--- auf Deutsch ---

Moin, IT Börner!

Wenn ich Dich richtig verstanden habe, beziehst Du dich vergleichsweise auf die unterlassenen Widerrufsbelehrungen im Shopping-TV. Das kann man leider nicht so ganz vergleichen, wobei ich auch nicht wirklich weiß, wie die das in der Praxis machen. Interessiert mich persönlich auch wenig, da ich – wie auch viele andere IT-Dienstleistungsunternehmen – anders handele.

Es ist perfide! Man stelle sich vor, man ist auf ausdrücklichen Wunsch des Kunden vor Ort wegen eines Problems mit dem Drucker. Während der Arbeiten stellt man fest, dass der PC des Weiteren potentiell gefährliche Software installiert hat, die aber mit dem eigentlichen Problem nichts zu tun hat. Will man diese dem Kunden jetzt (bezahlt!) im gleichen Termin entfernen, ist man dennoch gezwungen, ihm zunächst eine schriftliche Widerrufsbelehrung nebst Widerrufsfolgen- und Widerrufsausschlussbelehrung zu überreichen, die er mind. zwei mal unterzeichnen muss, bevor man rechtlich sichergehen kann, im Zweifelsfall auch seinen Lohn einklagen zu können. Wichtig ist halt, dass es VOR dem Vertragsschluss geschehen muss. Ansonsten kann der Kunde zurücktreten und muss nicht bezahlen. Denn der Lohn bzw. Wertersatz für die bereits geleistete Arbeit ist nur dann geschuldet, wenn vorher (!) korrekt aufgeklärt wurde.

Und jetzt zur Fernwartung. Wenn ich anbiete, via Telefon und Fernwartungssoftware regelmäßig Probleme auf Computern zu lösen, dann sind das Fernabsatzgeschäfte. Auch hier gilt: Bevor der Vertrag zustandekommt, muss der Kunde über seine Rechte und ggf. über den bewusst in Kauf genommenen Verlust ebendieser informiert werden. Und die Zurkenntnisnahme muss auch noch bestätigt werden. Ganz großartig, liebe EU!

**Reply by it-boerner** [reply]
I agree with you that this is a problem with the GDPR, but in the end you always have to have an AV contract because you do data processing with every remote maintenance. The same also applies to TV (TeamViewer).

If you set up a connection via TV, an AV contract must also be concluded beforehand or there must be approval. In my opinion, consent is given when the program is downloaded from the relevant operator.

If I offer my own client with my own server, then I am also responsible for it. So the person who downloads my client also agrees to my terms and privacy policy.

**Comment by rgigithub01** [comment]
We also looking for this feature. It would perfectly fit into the custom client generator in the Pro Version.
